### PR TITLE
[OTEL-953][opentelemetry] Increase backend timeout and ignore if unable to fetch from backend

### PR DESCRIPTION
**Pull request by songy23**
## Description

Increase backend timeout and ignore if unable to fetch from backend.

If backend provides traces / series / logs, still perform the validation.

If the tests themselves cannot be built and run, tests still fail.

## Motivation

Deflake opentelemetry tests due to staging not being stable.

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

Once your PR is reviewed, you can merge it! :heart:

## Reviewer checklist

* [ ] Check what scenarios are modified. If needed, add the relevant label (`run-parametric-scenario`, `run-profiling-scenario`...). If this PR modifies any system-tests internal, then add the `run-all-scenarios` label ([more info](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md)). 
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] if any of `build-some-image` label is present
  1. is the image labl have been updated ? 
  2. just before merging, locally build and push the image to hub.docker.com
* [ ] if a scenario is added (or removed), add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: tests/otel_tracing_e2e/test_e2e.py [hunk 1]
```diff
@@ -2,13 +2,16 @@
 import os
 import time
 
-from utils import context, weblog, interfaces, scenarios, irrelevant, flaky
-from utils.tools import get_rid_from_request
+from utils import context, weblog, interfaces, scenarios, irrelevant
+from utils.tools import get_logger, get_rid_from_request
```
Comment (cbeauchesne) [inline tests/otel_tracing_e2e/test_e2e.py:6]: a singleton is available: 

```suggestion
from utils.tools import logger, get_rid_from_request
```

File: tests/otel_tracing_e2e/test_e2e.py [hunk 2]
```diff
@@ -2,13 +2,16 @@
...
 from ._test_validator_trace import validate_all_traces
 from ._test_validator_log import validate_log, validate_log_trace_correlation
 from ._test_validator_metric import validate_metrics
 
 
+logger = get_logger()
```
Comment (cbeauchesne) [inline tests/otel_tracing_e2e/test_e2e.py:12]: You can remove this (see above)

File: tests/otel_tracing_e2e/test_e2e.py [hunk 3]
```diff
@@ -28,45 +31,49 @@ def test_main(self):
...
+                    dd_trace_id=dd_trace_id,
+                    dd_api_key=os.environ["DD_API_KEY_3"],
+                    dd_app_key=os.environ["DD_APP_KEY_3"],
+                )
+                for dd_trace_id in dd_trace_ids
+            ]
+
+        except ValueError:
+            logger.warning("Backend does not provide traces")
+            return
```
Comment (cbeauchesne) [inline tests/otel_tracing_e2e/test_e2e.py:70]: It 'll be the same as simply removing the test. Nobody will never look on this again, as the warning only shows in a log file.

If you want to give it a chance to be a real test on day, it's better to separate in a dedicated test function, and flag it as flaky (we're developping tooling to follow skipped tests).

File: tests/otel_tracing_e2e/test_e2e.py [hunk 4]
```diff
@@ -83,31 +90,36 @@ def setup_main(self):
...
+                    end=end,
+                    rid=rid,
+                    metric=metric,
+                    dd_api_key=os.environ["DD_API_KEY_3"],
+                    dd_app_key=os.environ["DD_APP_KEY_3"],
+                )
+                for metric in self.expected_metrics
+            ]
+
+        except ValueError:
```
Comment (cbeauchesne) [inline tests/otel_tracing_e2e/test_e2e.py:120]: same comment as above

File: tests/otel_tracing_e2e/test_e2e.py [hunk 5]
```diff
@@ -126,33 +138,41 @@ def test_main(self):
...
-            dd_app_key=os.environ["DD_APP_KEY_3"],
-        )
-        otel_log_trace_attrs = validate_log(log_collector, rid, "datadog_exporter")
-        trace_collector = interfaces.backend.assert_otlp_trace_exist(
-            request=self.r,
-            dd_trace_id=dd_trace_id,
-            dd_api_key=os.environ["DD_API_KEY_3"],
-            dd_app_key=os.environ["DD_APP_KEY_3"],
-        )
+        try:
```
Comment (cbeauchesne) [inline tests/otel_tracing_e2e/test_e2e.py:161]: Same comment

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

**Comment by songy23** [comment]
Friendly ping @cbeauchesne

**Comment by cbeauchesne** [comment]
Ah sorry, I understood from this comment that you'll split tests in two parts, the one validating backend, and the rest : 

> OK, separating the backend validation to another test sounds a better idea.
>
> With this change at least the CI failure indicates the otel tests cannot be built and should be fixed. We ran into similar issues before when breaking changes in this repo broke the otel tests and got ignored.

I misunderstood ?

**Comment by songy23** [comment]
> we're developping tooling to follow skipped tests

Sorry maybe I didn't read it correctly but do we already have the tooling in place?

**Comment by cbeauchesne** [comment]
> Sorry maybe I didn't read it correctly but do we already have the tooling in place?

Yes we have :)

**Comment by songy23** [comment]
Currently the OTel E2E tests heavily rely on the backend validation (in fact all validations are on backend data). Changing it will require several weeks' work. In the mean time I'd suggest merging this PR first, otherwise the CI will remain broken. WDYT?

**Comment by cbeauchesne** [comment]
Up to you, as long as you are ok with the risk of the test being not that useful, all good for me.

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**